test/Ocelot.AcceptanceTests/Steps.cs

```diff
@@ -1108,6 +1108,13 @@ public void ThenTheStatusCodeShouldBe(int expectedHttpStatusCode)
             responseStatusCode.ShouldBe(expectedHttpStatusCode);
         }
 
+        public void ThenRateLimitingHeadersExistInResponse(bool headersExist)
+        {
+            _response.Headers.Contains("X-Rate-Limit-Limit").ShouldBe(headersExist);
+            _response.Headers.Contains("X-Rate-Limit-Remaining").ShouldBe(headersExist);
+            _response.Headers.Contains("X-Rate-Limit-Reset").ShouldBe(headersExist);
```

Comment:
What about `Retry-After` header?
This header is mentioned in original issue #1305

Reply:
I've fixed Retry-After header in response.

Reply:
I still [don't see the new header](https://github.com/jlukawska/Ocelot/blob/fix/1305-ratelimiting-headers/test/Ocelot.AcceptanceTests/Steps.cs#L977-L982)! My understanding is that the header should be included in the upstream response headers to inform clients about the route state.

I'm curious if the acceptance tests are passing now.
🆗  I'll double-check that in the next code review phase.

Reply:
> I still [don't see the new header](https://github.com/jlukawska/Ocelot/blob/fix/1305-ratelimiting-headers/test/Ocelot.AcceptanceTests/Steps.cs#L977-L982)! My understanding is that the header should be included in the upstream response headers to inform clients about the route rate limit state.

The Retry-After header is sent with the response when rate limiting occured. It is sent to tell the client how long to wait before trying to send another HTTP request. That's why I don't check Retry-After existence when the response is OK.

> It is better to move this helper to the `ClientRateLimitTests` class. There is no sense to share this helper across all acceptance tests as a separate step in `Steps`.

I followed the current convention that I found in the tests, but sure, the Steps class grows too large and the tests should use it in a different way. I've changed that.